plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java

```diff
@@ -280,21 +284,29 @@ private String sshKeyAttributeName(String host, int isPrivate) {
 
     @SuppressWarnings("unchecked")
     ProfileDescriptor getUserProfile() throws ApiException, IOException, JsonParseException {
-        final String profileDescriptor = HttpJsonHelper.requestString(profileApiUrl, "GET", null);
-        return JsonHelper.fromJson(profileDescriptor, ProfileDescriptor.class, new TypeToken<ProfileDescriptor>() {}.getType());
+        final String profileDescriptor = HttpJsonHelper.requestString(profileApiUrl, HttpMethod.GET, null);
+        return DtoFactory.getInstance().createDtoFromJson(profileDescriptor, ProfileDescriptor.class);
     }
 
     @SuppressWarnings("unchecked")
     Map<String, String> getSshKeys() throws ApiException, IOException, JsonParseException {
-        final String preferencesJson = HttpJsonHelper.requestString(profileApiUrl + "/prefs?filter=ssh.key.*", "GET", null);
+        final String preferencesJson = HttpJsonHelper.requestString(profileApiUrl + "/prefs?filter=ssh.key.*", HttpMethod.GET, null);
         return JsonHelper.fromJson(preferencesJson, Map.class, new TypeToken<Map<String, String>>() {}.getType());
     }
 
     @SuppressWarnings("unchecked")
     void updateSshKeys(Map<String, String> preferences) throws ApiException, IOException {
-        HttpJsonHelper.requestString(profileApiUrl + "/prefs",
-                                     "POST",
-                                     JsonHelper.toJson(preferences),
-                                     Pair.of("Content-Type", "application/json"));
+        HttpJsonHelper.request(profileApiUrl + "/prefs",
+                               HttpMethod.POST,
+                               new JsonStringMapImpl<>(preferences),
+                               Pair.of("Content-Type", "application/json"));
+    }
+
+    @SuppressWarnings("unchecked")
+    void removeSshKey(List<String> keys) throws ApiException, IOException {
+        HttpJsonHelper.request(profileApiUrl + "/prefs",
+                               HttpMethod.DELETE,
+                               new JsonArrayImpl<>(keys),
+                               Pair.of("Content-Type", "application/json"));
```

Comment:
AFAIK it should be array of query params, not headers. BTW HTTPJsonHelper sends content type header already.